peerprep-fe/next.config.mjs

```diff
@@ -1,6 +1,7 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
   output: 'standalone',
+  reactStrictMode: false,
```

Comment:
Hard to unmount it because, there is an edge case where if the first listener (pre-rerender) receives the message, then the second rerender will not be able to pick it up (it will unmount the 1st listener, so the message will forever be gone)

I added useRef to ensure that it only creates the listener once